### PR TITLE
[rure] add recipe

### PR DESCRIPTION
**Pull request by chengchingwen**
This is the recipe for building the [c api of RUst's REgex engine](https://github.com/rust-lang/regex/tree/master/regex-capi)

### CODE REVIEW

**Comment by chengchingwen** [comment]
@giordano Could we merge this?

File: R/rure/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/rust-lang/regex.git", "061ee815ef2c44101dba7b0b124600fcb03c1912")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd regex/regex-capi/
```
Comment (giordano) [inline R/rure/build_tarballs.jl:16]:
```suggestion
cd $WORKSPACE/srcdir/regex/regex-capi/
```

File: R/rure/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+mkdir -p ${includedir}
+cp ./include/rure.h ${includedir}/rure.h
```
Comment (giordano) [inline R/rure/build_tarballs.jl:18]:
```suggestion
install -Dvm 644 include/rure.h "${includedir}/rure.h"
```

File: R/rure/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+cargo build --release
+cd ../
+install -Dvm 755 target/${rust_target}/release/*rure.${dlext} "${libdir}/librure.${dlext}"
```
Comment (giordano) [inline R/rure/build_tarballs.jl:21]:
```suggestion
install -Dvm 755 ../target/${rust_target}/release/*rure.${dlext} "${libdir}/librure.${dlext}"
```

**Pull request merged**